1-theory/1-apis-services-microservices.adoc

```diff
@@ -0,0 +1,144 @@
+____
+
+Service-orientated architecture is essentially the idea of taking various modules out of your application, and putting them on a different server, 
+with a network protocol in the way. Instead of having one application that handles user directories, billing, messaging, shipping, inventories, etc., 
+you have miltiple applications which handle one of those things independently.
+
+// TODO Diagram of example SoA, with a web app, mobile app, billing, user service, messaging service, etc
+
+There are a few benefits to SoA over creating one single giant application, so long as your team is qualified in handling the complications that come with it.
+
+One big example is reducing the effects of certain compliance restrictions. Financial standards like PCI might impose a lot of ruling around systems that store credit cards, so if the entire application has access to this data directly the whole application will need to be PCI compliant. If you split out the billing logic to its own service, then potentially only the billing service has to worry about being compliat.  
+
+Other compliance standards like SOX can impose strict rules on how deployments happen, multi-step review processes with multiple stakeholders signing off, who has access to what, etc. I´ve worked at 
+companies where SOX really slowed down deployments, meaning that even when code is finished and confirmed working, it might take a week or two to get it deployed. Other teams were happily cranking along, deploying multiple times a day, because the infection of SOX had not spread to their services. 
+
+Even if compliance isn´t an issue, comminicating over the network through standardized protocals make it possible for different
+services to be written in different programming languages or different frameworks. Instead of using the same language for everything, you 
+can use the right tool for the job, selecting a programming language that has strengths for certain tasks.
+
+WARNING: One thing to be wary of here is that some developers get a little giddy when told they can "use whatever language you like", finally giving CoolNewLanguage™ a try, even though the are not
+very experienced with it. Not only will that make it hard to maintain for that developer, it might also make it hard to find other developers
+who can help maintain it. If management are not careful, a company just getting into SoA can have 20 services in 20 different languages, and 
+this increases the danger of "Bus Factor" more than my 8 mile bike commute in NYC. 
+
+The positives are fairly clear, but an often overlooked problem with service-orientated arcitecture is the complexity of managing multiple services, and not just the comlpications of handling "more servers". Managing two services is not twice as hard as managing a single application, it is infinitely more complex, and you need people who know how to handle these issues. 
```

Comment:
Maybe mention the increase in difficulty of debugging issues that span multiple services. Also, the need for tooling and instrumentation that is harder to understand than being able to step through code with a debugger

Reply:
Got it!